### PR TITLE
#4 stdin support

### PR DESCRIPTION
**Pull request by sonnyp**

### CODE REVIEW

**Comment by sindresorhus** [comment]
I didn't mean it would just take the url/path from stdin, but rather the content, save it to a tempfile and then pass the tempfile path to `opn`.

**Comment by sindresorhus** [comment]
The `help` usage also needs to be updated in the cli.js and readme.

**Comment by sonnyp** [comment]
> I didn't mean it would just take the url/path from stdin, but rather the content, save it to a tempfile and then pass the tempfile path to opn.

yes, next PR

>  The help usage also needs to be updated in the cli.js and readme.

right, adding that

File: cli.js [hunk 1]
```diff
@@ -15,6 +18,7 @@ const cli = meow(`
 	  $ opn http://sindresorhus.com -- firefox
 	  $ opn http://sindresorhus.com -- 'google chrome' --incognito
 	  $ opn unicorn.png
+		$ echo http://sindresorhus.com | opn
```
Comment (sonnyp) [inline cli.js]: @sindresorhus good for you?

File: cli.js [hunk 2]
```diff
@@ -2,10 +2,13 @@
 'use strict';
 const meow = require('meow');
 const opn = require('opn');
+const getStdin = require('get-stdin');
 
 const cli = meow(`
 	Usage
 	  $ opn <file|url> [--wait] [-- <app> [args]]
+			or
+		$ stdout | opn [--wait] [-- <app> [args]]
```
Comment (sonnyp) [inline cli.js]: @sindresorhus good for you?

**Comment by sonnyp** [comment]
fixed indent

**Comment by SamVerschueren** [comment]
I miss this

``` js
if (input.length === 0 && process.stdin.isTTY) {
    console.error('Input required');
    process.exit(1);
}
```

Example: https://github.com/sindresorhus/sparkly-cli/blob/master/cli.js

**Comment by sonnyp** [comment]
@SamVerschueren I believe that should be an other PR

**Comment by sonnyp** [comment]
@sindresorhus review ?

**Pull request closed without merging**